module-4/backend/week_two.md

```diff
@@ -5,18 +5,57 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+
+ES6 has opened JS up to a bit more flexibility. Some of the syntax has changed, such as ability to use semicolons, and the declaration for variables has changed. 
+
 2. What's the difference between asynchronous and synchronous JavaScript? 
+
+Synchronous JS happens in a line, linearlly, all at once. Asynchronous JS doesn't necessarily do all the things listed on the page in order. 
+
 3. What are the four pillars of Object Oriented programming?
+
+Abstraction, Encapsulation, Inheritance, Polymorphism. 
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+
+The ability to declare variables and make objects and to pass information from one thing to another.
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+Be aware of repeating pieces of code. Try it in small steps. 
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+
+It is a function that is passed to another function. We can use these to string useful pieces together and make our code more dynamic. 
+
 7. What's the scope of variables in Javascript?
+
+There are global and local variable in JS. Global are those that can be accessed by the entire page of code while there are locals that are only availiable within a function. 
+
 8. What's the difference between `==` and `===` in JavaScript?
+
+I like to think of one as exactly equal and the other approximately equal. For instance '2' == 2 would evaluate to true, but '2' === 2 would evaluate to false. 
+
 9. Why do front end frameworks exist?
 
+So that we can have users interact with a webpage with minimal hassle. It allows us flexibility to display interesting things on a webpage for users to stay engaged. 
+
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
+
+HTTP is stateless because it doesn't need the server to keep session information for the duration of multiple requests. 
```

Comment:
http requests are independent. each request/response cycle knows nothing about the cycle that came before or the one that comes after. we use sessions to help keep state between these independent cycles in order for a user to stay logged in, etc.